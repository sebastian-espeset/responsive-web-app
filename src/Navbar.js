import {React, useState} from 'react';
import { StyledNavbar, 
        StyledLogoContainer, 
        StyledLogo, 
        LinksContainer, 
        NavbarInner, 
        LinksInnerContainer, 
        Link,
        OpenLink,
        LinksHamburgerButton,
        FindButtonLink,
        LoginButton,
        OpenFindButtonLink,
        OpenLoginButtonLink,
        NavbarMobileContainer 
      } from './styles/Navbar.style';

// Font for beam logo: Sofia Rouch Script
import logo from './assets/logo.png'

export default function Navbar() {
  const [openNav, setOpenNav]=useState(false);
  const [loginButtonClick, setLoginButtonClick]=useState(false);
  const [findButtonClick, setFindButtonClick]=useState(false);
  const [openLoginButtonClick, setOpenLoginButtonClick]=useState(false);
  const [openFindButtonClick, setOpenFindButtonClick]=useState(false);

  return (
    <StyledNavbar openNav = {openNav}>
      <NavbarInner>
        <StyledLogoContainer>
          <StyledLogo src={logo} alt="beam dental logo" data-testid='logo-test-id'/>
        </StyledLogoContainer>
        <LinksContainer>
          <LinksInnerContainer>
            <Link>Explore</Link>
            <Link>Company</Link>
            <Link>Resources</Link>

            {/* Would like to make a button component, would refactor nicely */}
            <FindButtonLink findButtonClick={findButtonClick} onClick={()=>{
              setFindButtonClick((val)=>!findButtonClick)
            }}>Find a dentist</FindButtonLink>

            <LoginButton loginButtonClick={loginButtonClick} onClick={()=>{
              setLoginButtonClick((val)=>!loginButtonClick)
            }}>Login</LoginButton>
            <LinksHamburgerButton 
            onClick={()=>{
              setOpenNav((val)=>!val)
            }}> 
            {openNav ? <>  &#10005;</>:<> &#8801;</> }
            </LinksHamburgerButton>
          </LinksInnerContainer>
        </LinksContainer>
      </NavbarInner>
      {openNav && (
        <NavbarMobileContainer>
          <OpenLink>Explore</OpenLink>
            <OpenLink>Company</OpenLink>
            <OpenLink>Resources</OpenLink>
            <OpenFindButtonLink openFindButtonClick={openFindButtonClick} onClick={()=>{
              setOpenFindButtonClick((val)=>!openFindButtonClick)
            }}>Find a dentist</OpenFindButtonLink>
            <OpenLoginButtonLink openLoginButtonClick={openLoginButtonClick} onClick={()=>{
              setOpenLoginButtonClick((val)=>!openLoginButtonClick)
            }}>Login</OpenLoginButtonLink>
        </NavbarMobileContainer>
      )}
    </StyledNavbar>
  )
}


        
