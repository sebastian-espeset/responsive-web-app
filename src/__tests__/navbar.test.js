import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';

afterEach(()=>{
    cleanup();
});

test('beam logo should render in the navbar',()=>{
    //Arrange
    render(<Navbar/>)
    //Act
    const wholeNavbar = screen.getByTestId('logo-test-id');
    //Assert
    expect(wholeNavbar).toBeInTheDocument();
});

// test will create a 'snapshot' of the html file, and then compare it to potential changes made    
test('navbar component to match snapshot',()=>{
    const tree = renderer.create(<Navbar/>).toJSON();
    expect(tree).toMatchSnapshot();
})