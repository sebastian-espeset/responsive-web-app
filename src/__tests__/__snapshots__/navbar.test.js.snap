// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar component to match snapshot 1`] = `
<nav
  className="sc-bczRLJ cUBFVg"
>
  <div
    className="sc-gsnTZi gzzSLJ"
  >
    <div
      className="sc-dkzDqf bBMfKQ"
    >
      <img
        alt="beam dental logo"
        className="sc-hKMtZM hqeGhs"
        data-testid="logo-test-id"
        src="logo.png"
      />
    </div>
    <div
      className="sc-eCYdqJ gVqsah"
    >
      <div
        className="sc-jSMfEi hwdenI"
      >
        <a
          className="sc-gKXOVf dVwIof"
        >
          Explore
        </a>
        <a
          className="sc-gKXOVf dVwIof"
        >
          Company
        </a>
        <a
          className="sc-gKXOVf dVwIof"
        >
          Resources
        </a>
        <button
          className="sc-ftvSup iJAGqv"
          onClick={[Function]}
        >
          Find a dentist
        </button>
        <button
          className="sc-papXJ hzJVgm"
          onClick={[Function]}
        >
          Login
        </button>
        <button
          className="sc-iqcoie ePikcJ"
          onClick={[Function]}
        >
           â‰¡
        </button>
      </div>
    </div>
  </div>
</nav>
`;
